tla-types/src/main/scala/at/forsyte/apalache/tla/typecmp/raw/ProtoBuilder.scala

```diff
@@ -0,0 +1,26 @@
+package at.forsyte.apalache.tla.typecmp.raw
+
+import at.forsyte.apalache.tla.lir.TlaEx
+import at.forsyte.apalache.tla.lir.oper.TlaOper
+import at.forsyte.apalache.tla.typecheck.etc.TypeVarPool
+import at.forsyte.apalache.tla.typecmp.{BuilderUtil, SignatureHandler}
+
+/**
+ * Base builder class that offers `simpleInstruction` to other builders.
+ *
+ * `simpleInstruction` creates a `buildInstruction` call, for an operator with a known signature.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+trait ProtoBuilder {
+  protected val varPool: TypeVarPool
+  protected val sigGen: SignatureHandler = new SignatureHandler(varPool)
+
+  /**
+   * Specialized `buildInstruction`, applicable when the TNT operator has a signature, that is, it is not overloaded. In
+   * that case, we fetch the computation from the [[SignatureHandler]].
+   */
+  protected def simpleInstruction(oper: TlaOper, args: TlaEx*): TlaEx =
```

Comment:
```suggestion
  protected def buildBySignatureLookup(oper: TlaOper, args: TlaEx*): TlaEx =
```

I think `simpleInstruction` is only meaningful in the eyes of the author. This method may construct operators that are not simple to me. Moreover, I (as a method user) don't really care whether it was simple to implement or not. It is probably important that such an expression is essentially constructed by a table lookup.